lib/sniffer.rb

```diff
@@ -45,12 +45,33 @@ def reset!
     end
 
     def store(data_item)
-      @data.add(data_item) if config.store
+      return unless config.store
+
+      if config.rotate?
+        rotate(data_item)
+      else
+        push(data_item) unless overflow?
+      end
     end
 
     def logger
       config.logger
     end
+
+    private
```

Comment:
I think all these private methods should be moved from `Sniffer` because they are about `Sniffer.data`.